lib/remap-lang/src/lib.rs

```diff
@@ -294,6 +294,46 @@ mod tests {
                 Err(r#"remap error: error for function "map_printer": cannot mark infallible function as "abort on error", remove the "!" signature"#),
                 Ok(().into()),
             ),
+            (
+                "$foo, $err = fallible_func!()",
+                Err(r#"remap error: assignment error: the variable "foo" does not need to handle the error-case, because its result is infallible"#),
```

Comment:
I left a comment in https://github.com/timberio/vector/issues/4803#issuecomment-756614824 to further improve this error message sometime next week when we're going to tackle compiler error messages.